### ISSUE TITLE
Distributed Pollers suddenly at 100% CPU and endless inceasing Log File

### ISSUE THREAD

**Issue by rsys-dev** [librenms/librenms#15423]
### The problem

Hello,

since the daily Update from 05.10. we have an Issue with our distributed pollers. CPU Load jumped to 100%, Memory Usage from 30% to ~95% and librenms.log growed to 8GB in 4 days, normally we have around 80MB in a week.
Our librenms.log is full of MPLS Module and SQL Errors. Especially the SQL error confuses me, since "ifMut" is not "datetime" and I can run the Query without any error directly on our MySQL Server.

I have no Idea, what causes this issue but I saw, that the poller was  updated on 04.10.

I modified "config/database.php" to set strict mode to false, that supressed the third error message `Syntax error or access violation: 1055 'librenms.s.device_id' isn't in GROUP BY`
I also tried `lnms config:set reporting.error false` mentioned in #15389 but it does not help (and it shouldn't since this issue was already fixed 3 days ago.

This error only ocures on all of our distributed pollers (we have 3) but not on the LibreNMS Server. 
Can anybody help me with this?

### Output of ./validate.php

```txt
===========================================
Component | Version
--------- | -------
LibreNMS  | 23.9.1-70-ga6239a733 (2023-10-09T02:00:05+02:00)
DB Schema | 2023_10_07_170735_increase_processes_cputime_length (261)
PHP       | 8.1.12
Python    | 3.7.3
Database  | MariaDB 10.3.36-MariaDB-0+deb10u2
RRDTool   | 1.7.1
SNMP      | 5.7.3
===========================================

[OK]    Composer Version: 2.6.4
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQl and PHP time match
[OK]    Distributed Polling setting is enabled globally
[OK]    Connected to rrdcached
[OK]    Active pollers found
[OK]    Dispatcher Service is enabled
[OK]    Locks are functional
[OK]    Python wrapper cron entry is not present
[OK]    Redis is functional
[OK]    rrdtool version ok
[OK]    Connected to rrdcached
[WARN]  Your local git contains modified files, this could prevent automatic updates.
        [FIX]:
        You can fix this with ./scripts/github-remove
        Modified Files:
         config/database.php
         rrd/.gitignore
```

### What was the last working version of LibreNMS?

Unknown, before 05.10. 02:00 CEST

### Anything in the logs that might be useful for us?

```txt
Error polling mpls module for nokia-7250-sw-a. ErrorException: Undefined array key "vRtrMplsLspFromAddr" in /opt/librenms/LibreNMS/OS/Timos.php:491
Stack trace:
#0 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(254): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}()
#2 /opt/librenms/app/Providers/ErrorReportingProvider.php(199): call_user_func()
#3 /opt/librenms/LibreNMS/OS/Timos.php(491): App\Providers\ErrorReportingProvider->handleError()
#4 /opt/librenms/LibreNMS/Modules/Mpls.php(122): LibreNMS\OS\Timos->pollMplsLsps()
#5 /opt/librenms/LibreNMS/Poller.php(191): LibreNMS\Modules\Mpls->poll()
#6 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#7 /opt/librenms/app/Console/Commands/DevicePoll.php(44): LibreNMS\Poller->poll()
#8 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#9 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#11 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod()
#12 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#13 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(208): Illuminate\Container\Container->call()
#14 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#15 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(178): Symfony\Component\Console\Command\Command->run()
#16 /opt/librenms/vendor/symfony/console/Application.php(1081): Illuminate\Console\Command->run()
#17 /opt/librenms/vendor/symfony/console/Application.php(320): Symfony\Component\Console\Application->doRunCommand()
#18 /opt/librenms/vendor/symfony/console/Application.php(174): Symfony\Component\Console\Application->doRun()
#19 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#20 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#21 /opt/librenms/lnms(40): App\Console\Kernel->handle()
#22 {main}
Undefined array key "vRtrMplsLspFromAddr" {"exception":"[object] (ErrorException(code: 0): Undefined array key \"vRtrMplsLspFromAddr\" at /opt/librenms/LibreNMS/OS/Timos.php:491)"}

SQLSTATE[22007]: Invalid datetime format: 1366 Incorrect integer value: 'NULL' for column `librenms`.`ports`.`ifMtu` at row 1 (Connection: mysql, SQL: UPDATE `ports` set `ifMtu`=NULL,`ifInOctets`=0,`ifInOctets_prev`=0,`ifInOctets_rate`=0,`ifInOctets_delta`=0,`ifOutOctets`=0,`ifOutOctets_prev`=0,`ifOutOctets_rate`=0,`ifOutOctets_delta`=0,`ifInErrors`=0,`ifInErrors_prev`=0,`ifInErrors_rate`=0,`ifInErrors_delta`=0,`ifOutErrors`=0,`ifOutErrors_prev`=0,`ifOutErrors_rate`=0,`ifOutErrors_delta`=0,`ifInUcastPkts`=0,`ifInUcastPkts_prev`=0,`ifInUcastPkts_rate`=0,`ifInUcastPkts_delta`=0,`ifOutUcastPkts`=0,`ifOutUcastPkts_prev`=0,`ifOutUcastPkts_rate`=0,`ifOutUcastPkts_delta`=0,`poll_time`=1696850071,`poll_prev`=1696850054,`poll_period`=17 WHERE `port_id` = 233391) (Connection: dbFacile, SQL: UPDATE `ports` set `ifMtu`=NULL,`ifInOctets`=0,`ifInOctets_prev`=0,`ifInOctets_rate`=0,`ifInOctets_delta`=0,`ifOutOctets`=0,`ifOutOctets_prev`=0,`ifOutOctets_rate`=0,`ifOutOctets_delta`=0,`ifInErrors`=0,`ifInErrors_prev`=0,`ifInErrors_rate`=0,`ifInErrors_delta`=0,`ifOutErrors`=0,`ifOutErrors_prev`=0,`ifOutErrors_rate`=0,`ifOutErrors_delta`=0,`ifInUcastPkts`=0,`ifInUcastPkts_prev`=0,`ifInUcastPkts_rate`=0,`ifInUcastPkts_delta`=0,`ifOutUcastPkts`=0,`ifOutUcastPkts_prev`=0,`ifOutUcastPkts_rate`=0,`ifOutUcastPkts_delta`=0,`poll_time`=1696850071,`poll_prev`=1696850054,`poll_period`=17 WHERE `port_id` = 233391)#0 /opt/librenms/includes/polling/ports.inc.php(935): dbUpdate()
#1 /opt/librenms/LibreNMS/Modules/LegacyModule.php(123): include('...')
#2 /opt/librenms/LibreNMS/Poller.php(191): LibreNMS\Modules\LegacyModule->poll()
#3 /opt/librenms/LibreNMS/Poller.php(103): LibreNMS\Poller->pollModules()
#4 /opt/librenms/app/Console/Commands/DevicePoll.php(44): LibreNMS\Poller->poll()
#5 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#6 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#7 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#8 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod()
#9 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(208): Illuminate\Container\Container->call()
#11 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#12 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(178): Symfony\Component\Console\Command\Command->run()
#13 /opt/librenms/vendor/symfony/console/Application.php(1081): Illuminate\Console\Command->run()
#14 /opt/librenms/vendor/symfony/console/Application.php(320): Symfony\Component\Console\Application->doRunCommand()
#15 /opt/librenms/vendor/symfony/console/Application.php(174): Symfony\Component\Console\Application->doRun()
#16 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#17 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#18 /opt/librenms/lnms(40): App\Console\Kernel->handle()
#19 {main}

SQLSTATE[42000]: Syntax error or access violation: 1055 'librenms.s.device_id' isn't in GROUP BY (Connection: mysql, SQL: SELECT devices.display as displayName, sub.* FROM devices, ( SELECT s.svc_oid, s.device_id, count(s.svc_oid) as idCount, count(DISTINCT s.svcDescription) as descriptionDuplicates, sum(IF(s.svcDescription = '', 1, 0)) as emptyDescription FROM `mpls_services` s GROUP BY s.svc_oid HAVING (idCount > 1 AND descriptionDuplicates > 1) OR emptyDescription > 0 ) as sub WHERE devices.device_id = 2317 AND sub.device_id = devices.device_id) (Connection: dbFacile, SQL: SELECT devices.display as displayName, sub.* FROM devices, ( SELECT s.svc_oid, s.device_id, count(s.svc_oid) as idCount, count(DISTINCT s.svcDescription) as descriptionDuplicates, sum(IF(s.svcDescription = '', 1, 0)) as emptyDescription FROM `mpls_services` s GROUP BY s.svc_oid HAVING (idCount > 1 AND descriptionDuplicates > 1) OR emptyDescription > 0 ) as sub WHERE devices.device_id = 2317 AND sub.device_id = devices.device_id)#0 /opt/librenms/LibreNMS/Alert/AlertRules.php(72): dbFetchRows()
#1 /opt/librenms/app/Actions/Alerts/RunAlertRulesAction.php(53): LibreNMS\Alert\AlertRules->runRules()
#2 /opt/librenms/app/Action.php(39): App\Actions\Alerts\RunAlertRulesAction->execute()
#3 /opt/librenms/app/Listeners/CheckAlerts.php(32): App\Action::execute()
#4 /opt/librenms/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php(441): App\Listeners\CheckAlerts->handle()
#5 /opt/librenms/vendor/laravel/framework/src/Illuminate/Events/Dispatcher.php(249): Illuminate\Events\Dispatcher->Illuminate\Events\{closure}()
#6 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(433): Illuminate\Events\Dispatcher->dispatch()
#7 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Events/Dispatchable.php(14): event()
#8 /opt/librenms/LibreNMS/Poller.php(134): App\Events\DevicePolled::dispatch()
#9 /opt/librenms/app/Console/Commands/DevicePoll.php(44): LibreNMS\Poller->poll()
#10 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#11 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#12 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#13 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod()
#14 /opt/librenms/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#15 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(208): Illuminate\Container\Container->call()
#16 /opt/librenms/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#17 /opt/librenms/vendor/laravel/framework/src/Illuminate/Console/Command.php(178): Symfony\Component\Console\Command\Command->run()
#18 /opt/librenms/vendor/symfony/console/Application.php(1081): Illuminate\Console\Command->run()
#19 /opt/librenms/vendor/symfony/console/Application.php(320): Symfony\Component\Console\Application->doRunCommand()
#20 /opt/librenms/vendor/symfony/console/Application.php(174): Symfony\Component\Console\Application->doRun()
#21 /opt/librenms/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#22 /opt/librenms/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#23 /opt/librenms/lnms(40): App\Console\Kernel->handle()
#24 {main}
```

**Comment by murrant** [comment]
Third error is a bad alert rule you'll need to fix.

**Comment by rsys-dev** [comment]
> Third error is a bad alert rule you'll need to fix.

Thanks for the note, we have identified the incorrect alert rule and are working to fix the issue.
We have also found the cause that the pollers were running at 100% CPU utilization. With a system update PHP 8.2 was installed and apparently the php.ini of PHP 8.1 was not taken over correctly, which caused the timezone of the pollers to differ from the timezone of the SQL Server. This has been corrected and the CPU load has returned to a normal level.

We also found other error messages from the TiMOS MPLS module, but I will open a new issue for this, as this has nothing to do with the topic of this issue.
